catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.java

```diff
@@ -123,6 +130,25 @@ public boolean onOptionsItemSelected(MenuItem item) {
 		return super.onOptionsItemSelected(item);
 	}
 
+	public static boolean isRecoveredFromCrash() {
+		return sharedPreferences != null && sharedPreferences.getBoolean(RECOVERED_FROM_CRASH, false);
+	}
+
+	public static boolean checkIfCrashRecoveryAndFinishActivity(final Activity context) {
+		if (isRecoveredFromCrash()) {
+
+			CrashReporter.sendUnhandledCaughtException();
```

Comment:
I would like to move this line to `onCreate`.
This removes the dependency between crash recovering and crash reporting entirely.

So the CrashReporter is like: "I don't care if the activity is recovering from a crash or not. If there is an exception stored, I'll report it."
And the CrashRecovering is like: "I don't care about the CrashReporter at all. I'll just finish every activity till I'm back in the MainMenuActivity."